src/lib.rs

```diff
@@ -622,8 +670,19 @@ impl<T> SliceDeque<T> {
     }
 
     /// Moves the deque head by `x`.
+    ///
+    /// # Panics
+    ///
+    /// If the `head` wraps over the `tail` the behavior is undefined, that is,
+    /// if `x > 0 && x > size()` or if `x < 0 && x < -(capacity() - size())`.
+    /// If `-C debug-assertions=1` violating this pre-condition `panic`s.
```

Comment:
Does it cause performance regressions to retain these asserts in release mode? If you're checking the length before adding or removing an element then the optimizer should be able to elide them but they're good sanity checks for manual manipulation.

Reply:
> Does it cause performance regressions to retain these asserts in release mode?

These methods were only used internally inside `SliceDeque` and there everywhere where they are used properly upholds these pre-conditions without any checks.

I can rename these to `move_{head,tail}_unchecked` and add checked variants, but the checked variants must still be `unsafe` anyways, so I don't see that much win doing this. 

I'll think about it today.

Reply:
It's up to you. I try to cover as much as my unsafe code with asserts as I can, for sanity and bug-checking.

Reply:
I've added `move_{head,tail}_unchecked` in the latest commit, and `move_{head,tail}` now always check these pre-conditions :)